clientapi/routing/membership.go

```diff
@@ -104,23 +106,39 @@ func SendMembership(
 		return jsonerror.InternalServerError()
 	}
 
-	if _, err := producer.SendEvents(
-		req.Context(),
-		[]gomatrixserverlib.HeaderedEvent{(*event).Headered(verRes.RoomVersion)},
-		cfg.Matrix.ServerName,
-		nil,
-	); err != nil {
-		util.GetLogger(req.Context()).WithError(err).Error("producer.SendEvents failed")
-		return jsonerror.InternalServerError()
-	}
-
 	var returnData interface{} = struct{}{}
 
-	// The join membership requires the room id to be sent in the response
-	if membership == gomatrixserverlib.Join {
+	switch membership {
+	case gomatrixserverlib.Invite:
+		// Invites need to be handled specially
+		err = producer.SendInvite(
```

Comment:
We still fall through to `producer.SendEvents(` afterwards which feels like a bug?

Reply:
At the moment this is deliberate bceause `producer.SendInvites()` only notifies the federation sender/sync API about the invites but doesn't update the actual room state itself - I am still thinking about the best way to solve this so we don't need to do both.